lib/pbench/agent/tool_meister.py

```diff
@@ -1045,6 +1056,10 @@ def init_tools(self, data):
                 else:
                     self._persistent_tools[name] = persistent_tool
                     self.logger.debug("NAME: " + name + "  TOOL OPTS: " + tool_opts)
+
+        # Remember this persistent tmp tool directory so that we can delete it
+        # when requested.
+        self.directories[data["directory"]] = _tool_dir
```

Comment:
> And, why is it using `data["directory"]` as a key? Is that value meaningful when `self._controller` is not the same as `self._hostname`?

`data["directory"]` is a value where the persistent tool data is going to be stored on the controller. The key is always (in case of persistent tools) `/var/lib/pbench-agent/<benchmark-name>/tools-<group_name>` irrespective of whether hostname is same as controller. The value for this key is either a remote tmp directory in case there is a remote or same as key in case hostname is the controller. The key just points to where the tmp tools data is stored.